### PR TITLE
Minor parser clean-up

### PR DESCRIPTION
**Pull request by lukehsiao**

### CODE REVIEW

**Comment by lukehsiao** [comment]
I split of the changes that were unrelated to the big parser.py refactor so that we can merge in some of this cleanup before the bigger change.

File: fonduer/candidates/models/__init__.py [hunk 1]
```diff
@@ -2,12 +2,14 @@
...
 from fonduer.candidates.models.implicitspan import ImplicitSpan, TemporaryImplicitSpan
 from fonduer.candidates.models.span import Span, TemporarySpan
+from fonduer.candidates.models.temporarycontext import TemporaryContext
 
 __all__ = [
     "Candidate",
     "Image",
     "ImplicitSpan",
     "Span",
+    "TemporaryContext",
```
Comment (lukehsiao) [inline fonduer/candidates/models/__init__.py]: Given that this is related to candidate gen, moving over to candidates directory.

File: fonduer/parser/models/phrase.py [hunk 2]
```diff
@@ -189,12 +189,12 @@ def __repr__(self):
...
                 else self.col_start
             )
-            return "Phrase (testDoc: '{}', Table: {}, Row: {}, Col: {}, Index: {}, Text: '{}')".format(
+            return "Phrase (Doc: '{}', Table: {}, Row: {}, Col: {}, Index: {}, Text: '{}')".format(
                 self.document.name,
                 self.table.position,
                 rows,
                 cols,
-                self.position,
+                self.phrase_num,
```
Comment (lukehsiao) [inline fonduer/parser/models/phrase.py]: Eliminating the redundancy of position/phrase_num.

File: fonduer/parser/simple_tokenizer.py [hunk 3]
```diff
@@ -9,7 +9,7 @@
 class SimpleTokenizer(object):
     """Tokenizes text on whitespace only using split()."""
 
-    def __init__(self, delim):
+    def __init__(self, delim="<NB>"):
```
Comment (lukehsiao) [inline fonduer/parser/simple_tokenizer.py]: This allows us to eliminate the duplicated `self.delim = "<NB>"` when we refactor the parser later.

File: fonduer/candidates/models/__init__.py (hunk 1 above)
Comment (SenWu) [reply fonduer/candidates/models/__init__.py]: 👍

File: fonduer/parser/models/phrase.py (hunk 2 above)
Comment (SenWu) [reply fonduer/parser/models/phrase.py]: 👍

**Pull request merged**